contracts/PBAB+Collabs/GenArt721CoreV2_PBAB_FLEX.sol

```diff
@@ -0,0 +1,862 @@
+        projects[_projectId].scriptCount = projects[_projectId].scriptCount + 1;
+    }
+
+    /**
+     * @notice Updates script for project `_projectId` at script ID `_scriptId`.
+     * @param _projectId Project to be updated.
+     * @param _scriptId Script ID to be updated.
+     * @param _script Script to be added.
+     */
+    function updateProjectScript(
+        uint256 _projectId,
+        uint256 _scriptId,
+        string memory _script
+    ) public onlyUnlocked(_projectId) onlyArtistOrWhitelisted(_projectId) {
+        require(
+            _scriptId < projects[_projectId].scriptCount,
+            "scriptId out of range"
+        );
+        projects[_projectId].scripts[_scriptId] = _script;
+    }
+
+    /**
+     * @notice Updates external asset dependency for project `_projectId`.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index.
+     * @param _cid Asset cid.
+     * @param _dependencyType Asset dependency type.
+     */
+    function updateProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index,
+        string memory _cid,
+        ExternalAssetDependencyType _dependencyType
+    )
+        public
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projectIdToExternalAssetDependencies[_projectId].length,
+            "Asset index out of range"
+        );
+        projectIdToExternalAssetDependencies[_projectId][_index].cid = _cid;
+        projectIdToExternalAssetDependencies[_projectId][_index]
+            .dependencyType = _dependencyType;
+        emit ExternalAssetDependencyUpdated(
+            _projectId,
+            _index,
+            _cid,
+            _dependencyType
+        );
+    }
+
+    /**
+     * @notice Removes external asset dependency for project `_projectId`.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index
+     */
+    function removeProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index
+    )
+        public
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projectIdToExternalAssetDependencies[_projectId].length,
+            "Asset index out of range"
+        );
+
+        string memory _cid = projectIdToExternalAssetDependencies[_projectId][
+            _index
+        ].cid;
+        ExternalAssetDependencyType _dependencyType = projectIdToExternalAssetDependencies[
+                _projectId
+            ][_index].dependencyType;
+
+        for (
+            uint256 i = _index;
+            i < projectIdToExternalAssetDependencies[_projectId].length - 1;
+            i++
+        ) {
+            projectIdToExternalAssetDependencies[_projectId][
+                i
+            ] = projectIdToExternalAssetDependencies[_projectId][i + 1];
+        }
```

Comment:
(if we decide to use a mapping instead of array, can perform the same general operation of moving last index to index being deleted, then delete last index.)

Reply:
discussed async in DMs but came to the conclusion that order really doesn't matter for us, will move forward with these changes